examples/experimental/dagster-airlift/dagster_airlift/test/airflow_test_instance.py

```diff
@@ -0,0 +1,131 @@
+from collections import defaultdict
+from datetime import datetime
+from typing import Dict, List, Optional, Tuple
+
+import requests
+
+from dagster_airlift.core import AirflowInstance
+from dagster_airlift.core.airflow_instance import DagInfo, DagRun, TaskInfo, TaskInstance
+from dagster_airlift.core.basic_auth import AirflowAuthBackend
+
+
+class DummyAuthBackend(AirflowAuthBackend):
+    def get_session(self) -> requests.Session:
+        raise NotImplementedError("This shouldn't be called from this mock context.")
+
+    def get_webserver_url(self) -> str:
+        return "http://dummy.domain"
+
+
+class AirflowTestInstance(AirflowInstance):
```

Comment:
Can we call this `AirflowInstanceFake` or something?

Also as follow up let's consider adding an `ABC` `IAirflowInstance` that is stateless. It's quite problematic to inherit from the fully featured class in this way.